### ISSUE TITLE
Creating a new tag

### ISSUE THREAD

**Issue by garymh** [BenjaminOddou/alfred-emoji-wine#8]
Hi - this might be a dumb question, but I'm a little confused on how you enter the tag section. Typing the `harvest` keyword brings up:

<img width="568" alt="Screenshot 2024-02-26 at 20 32 54" src="https://github.com/BenjaminOddou/alfred-emoji-wine/assets/235202/e7194ae5-13b6-4442-a454-d8782b48f538">

but after that, hitting `return` on the Tags section just closes the Alfred window. Typing any arguments (`harvest tag`) just goes to my default search...

I'm using Python 3.12.2 and pillow 10.2.0_1

**Comment by BenjaminOddou** [comment]
Hi @garymh 

I'll try to figure out what's causing the issue. Could you give me the Alfred logs ?

**Comment by BenjaminOddou** [comment]
really strange, could you confirm me that you have version 1.3.0 of the workflow ?

When I trigger harvest :

```log
[21:55:42.438] Logging Started...
[21:55:46.690] Emoji Wine[Script Filter] Queuing argument '(null)'
[21:55:46.878] Emoji Wine[Script Filter] Script with argv '(null)' finished
[21:55:46.879] Emoji Wine[Script Filter] {"items": [{"title": "Refresh the API", "subtitle": "Press \u23ce to grab emoji data with language : \ud83c\uddec\ud83c\udde7 English", "arg": "_api", "icon": {"path": "icons/sync.webp"}}, {"title": "Tags", "subtitle": "Create, modify or remove tags for \ud83c\uddec\ud83c\udde7 English", "arg": "_rerun;;1;;", "icon": {"path": "icons/tag.webp"}}, {"title": "API information", "subtitle": "Last update : 10-02-2024 18:16:56 \u01c0 3782 Emojis \u01c0 \ud83c\uddec\ud83c\udde7 English", "valid": false, "icon": {"path": "icons/info.webp"}}]}
[21:55:48.515] Emoji Wine[Script Filter] Processing complete
[21:55:48.517] Emoji Wine[Script Filter] Passing output '_rerun;;1;;' to Split Arg
[21:55:48.518] Emoji Wine[Split Arg] Processing complete
[21:55:48.520] Emoji Wine[Split Arg] Passing output '_rerun;;1;;' to Conditional
[21:55:48.522] Emoji Wine[Conditional] Processing complete
[21:55:48.524] Emoji Wine[Conditional] Passing output '_rerun;;1;;' to Call External Trigger
[21:55:48.525] Emoji Wine[External] Processing complete
[21:55:48.527] Emoji Wine[External] Passing output '_rerun;;1;;' to Arg and Vars
[21:55:48.543] Emoji Wine[Arg and Vars] Processing complete
[21:55:48.545] Emoji Wine[Arg and Vars] Passing output '' to Script Filter
[21:55:48.547] Emoji Wine[Script Filter] Queuing argument '(null)'
[21:55:48.759] Emoji Wine[Script Filter] Script with argv '(null)' finished
[21:55:48.766] Emoji Wine[Script Filter] {"items": [{"title": "Return", "subtitle": "Back to previous state", "arg": "_rerun;;0;;", "icon": {"path": "icons/return.webp"}}, {"title": "Add a new tag", "subtitle": "Create a new tag and assign it to emojis", "arg": "_tags;new;;;_rerun\u01c0\u01c01\u01c0\u01c0", "icon": {"path": "icons/new.webp"}}, {"title": "Delete a tag", "subtitle": "Erase a tag from the list below", "arg": "_rerun;delete;2;;", "icon": {"path": "icons/delete.webp"}}]}
```

I am seeing this :

![Capture d’écran 2024-02-27 à 22 11 04](https://github.com/BenjaminOddou/alfred-emoji-wine/assets/85166574/9185e88d-d70d-4ab5-9429-ee9de7568b70)

From there you can create tags and associate it to emojis.

**Comment by BenjaminOddou** [comment]
could you check that you have `tags-en.json` file here : `~/Library/Application Support/Alfred/Workflow Data/com.benjamino.emoji_wine`and also please check for any error in the workflow logs located here `~/Library/Caches/com.runningwithcrayons.Alfred/Workflow Data/com.benjamino.emoji_wine`

**Comment by BenjaminOddou** [comment]
Could you try to refresh your API ?

**Comment by BenjaminOddou** [comment]
Yes I think something is not configured correctly but I can't really know what's happening. How did you install python ? Could you also give me the new logs ? Could you maybe try to reinstall the workflow (just to make sure). I'll try my best to debug.

**Comment by BenjaminOddou** [comment]
Ok then. Could you confirm me that you don't have any emojis icons when you trigger `wine` ? There might be something with pillow but as you seems to have the latest version 🤔. Could you also check that you have the emoji font here `/System/Library/Fonts/Apple Color Emoji.ttc`. I put that raw meaning I expect it's always available under this path, but maybe on newer Macs (I have an intel chip) the path is different ?

**Comment by BenjaminOddou** [comment]
mhh kind of weird. Is it possible that the emojis without icons are emojis from unicode 15.1 ? There aren't available yet but the unicode is already in the api from end of 2023 (see this [article](https://blog.emojipedia.org/first-look-new-emojis-in-ios-17-4/)). But it shouldn't be a problem, for me it looks like this:

![mushroom](https://github.com/BenjaminOddou/alfred-emoji-wine/assets/85166574/a04e3aa7-5f2a-41df-a4af-21994ab4c8f2)

Could you delete these elements  from this folder `~/Library/Caches/com.runningwithcrayons.Alfred/Workflow Data/com.benjamino.emoji_wine` before refreshing the API ?

![cache](https://github.com/BenjaminOddou/alfred-emoji-wine/assets/85166574/0e2ae01c-c6aa-4e86-b74a-206e0d0512ee)

**Comment by BenjaminOddou** [comment]
I can fix that by just ignoring emojis that fails to be converted into images, or assign them a default image. Also, is there any improvement in the tags section (that was the original subject 🙃)? 

Yes some of the codes can be "merged" as one emoji (👋 + 🏻 = 👋🏻, see [here](https://emojipedia.org/waving-hand-light-skin-tone)). When the Unicode consortium approved an emoji, each platform (Google, Facebook, Apple)  can integrate these codes with a design. Note that the design can change over the years but not the codes.

What I still find weird is : why the behavior is different ? Maybe we can create a GitHub issue in the pillow repo. I can do that if you can provide : 

- The python version (3.12.2) ?
- The pillow version (10.2.0) ?
- The MacOS version
- Your chip

**Comment by BenjaminOddou** [comment]
@garymh it should be fix in the [v1.4.0](https://github.com/BenjaminOddou/alfred-emoji-wine/releases/tag/v1.4.0). Emojis that fails to be converted should have this icon :

![fail](https://github.com/BenjaminOddou/alfred-emoji-wine/assets/85166574/1fa67b89-1e70-406c-b729-c9c90bcbadec)
